### ISSUE TITLE
Cannot create a release with 1.4.3

### ISSUE THREAD

**Issue by href** [lgiordani/punch#31]
The latest release seems to have broken punch.py for us.

Running `punch --action build` fails as follows:

```python-traceback
Traceback (most recent call last):
  File "/Users/denis/.virtualenvs/onegov/bin/punch", line 11, in <module>
    sys.exit(main())
  File "/Users/denis/.virtualenvs/onegov/lib/python3.6/site-packages/punch/cli.py", line 295, in main
    uc.finish_release()
  File "/Users/denis/.virtualenvs/onegov/lib/python3.6/site-packages/punch/vcs_repositories/git_repo.py", line 99, in finish_release
    self._run(command_line)
  File "/Users/denis/.virtualenvs/onegov/lib/python3.6/site-packages/punch/vcs_repositories/vcs_repo.py", line 65, in _run
    raise RepositorySystemError(error_message)
punch.vcs_repositories.exceptions.RepositorySystemError: An error occurred executing 'git commit -m Release 2018.3.15':
Process output was: On branch 2018.3.15
Changes not staged for commit:
	modified:   punch_version.py
	modified:   setup.py

no changes added to commit
```

This seems to happen because the files are no longer added due to this change:
https://github.com/lgiordani/punch/commit/1cc1754a4f4e083f2d75b9f4c2d97c252ec34d56

What's the reason for this being commented out?

**Comment by lgiordani** [comment]
Sorry for the issue, that was commented (actually I wanted to remove it, I left it commented by mistake) because it was silently adding non-tracked files. I'll try to solve your issue as soon as possible.
Sorry again for the trouble

**Comment by href** [comment]
Well you might have to add some files I think ;)

Anyway, no worries - take your time. We'll simply skip 1.4.3.

**Comment by lgiordani** [comment]
May you please post your `punch_config.py` and `punch_version.py`? This way I have a proper test case. Thanks

**Comment by href** [comment]
You can clone the following repository to test:
https://github.com/onegov/onegov.applications

**Comment by jobec** [comment]
Bumped into the same thing today. If you change files because you updated the version in them, you of course have to add them eventually.

I actually used this behavior...  
I edit my changelog manually, then change the version and have the changelog end up in the same commit.

Maybe make this configurable?
By default only `git add` the files defined in `punch_config.py`.
And an option to add ALL files.

**Comment by lgiordani** [comment]
Hi! I think I fixed the issue. The develop branch contains a version that should work for both of you.
@href punch now commits its config and version files plus all the files it manages
@jobec the VCS dict in the config file accepts an `include_files` list of files that have to be included in the commit. It doesn't allow to add all the files, still. Let me know if you think this is good enough.

**Comment by lgiordani** [comment]
Ok, in the end I opted for another configuration flag `include_all_files`, mostly because personally I don't like to have something that can be either a list or a string. I'm going to release a new version soon, then I'll close this issue

**Comment by lgiordani** [comment]
I released version 1.5.0 that fixes this issue. Thanks for the collaboration!

**Comment by jobec** [comment]
I can confirm it works. 👍
